src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -298,4 +301,77 @@ private function getPrecisionFromCart(Cart $cart): int
 
         return $computingPrecision->getPrecision((int) $currency->precision);
     }
+
+    /**
+     * Update order details after a specific price has been created or updated
+     *
+     * @param Order $order
+     * @param OrderDetail $updatedOrderDetail
+     * @param Product $product
+     * @param int|null $combinationId
+     * @param int $productQuantity
+     * @param Number $priceTaxIncluded
+     * @param Number $priceTaxExcluded
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function updateOrderDetailsWithSameProduct(
```

Comment:
This method along with `updateOrderInvoices` must not be public It's a private method that is called by the `update` method
This way the service only has one end point, and it manages the complexity internally